### PR TITLE
miniAODIsolation_91X_V2

### PR DESCRIPTION
**Pull request by archiron**
The electron MC miniAOD validation implements a PF isolation recomputation to ensure that no information is lost when producing the miniAODs.

The PF isolation as computed in RECO has been modified/updated (see
https://github.com/cms-sw/cmssw/commit/455a974a1f21d08cadc9bcc029cbfefab33d841c)
and the miniAOD computation was no longer in sync with the RECO.
The disagreement can be seen here:
http://cms-egamma.web.cern.ch/cms-egamma/validation/Electrons/Releases/9_1_0_pre2_miniAOD_DQM_std/GedvsGed_9_1_0_pre2_miniAOD_DQM_std/Fullgedvsged_RelValSingleElectronPt10_gedGsfE_Startup/#h_ele_chargedHadronRelativeIso_mAOD_recomp
and it was spotted in the validation, see for example:
https://hypernews.cern.ch/HyperNews/CMS/get/relval/7317/11.html

This PR implements a fix in the miniaod electron validation as well as an improvement: in principle any future change in RECO will be automatically picked up in the miniAOD validation.

Changes have been made in :
Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc
Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h
for the producer,

Validation/RecoEgamma/python/ElectronMcSignalValidatorMiniAOD_cfi.py
Validation/RecoEgamma/python/electronValidationSequenceMiniAOD_cff.py
for the isolation module used,

Validation/RecoEgamma/test/ElectronMcSignalValidationMiniAOD_cfg.py
for config file.

The effect can be observed here:
http://cmsdoc.cern.ch/cms/Physics/egamma/www/validation/Electrons/Dev/9_1_0_pre2_miniAOD_9_DQM_dev/GedvsGed_9_1_0_pre2_DQM_dev/Fullgedvsged_RelValSingleElectronPt10_gedGsfE_Startup/#h_ele_chargedHadronRelativeIso_mAOD_recomp

@beaudett @fcouderc

### CODE REVIEW

**Comment by dmitrijus** [comment]
+1

File: Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc [hunk 1]
```diff
@@ -15,13 +15,16 @@ typedef edm::Ptr<pat::Electron> PatElectronPtr;
 ElectronMcSignalValidatorMiniAOD::ElectronMcSignalValidatorMiniAOD(const edm::ParameterSet& iConfig) : ElectronDqmAnalyzerBase(iConfig)
 {
     mcTruthCollection_ = consumes<edm::View<reco::GenParticle> >(iConfig.getParameter<edm::InputTag>("mcTruthCollection")); // prunedGenParticles
-    electronToken_ = consumes<pat::ElectronCollection>(iConfig.getParameter<edm::InputTag>("electrons")); // slimmedElectrons
+    electronToken_     = consumes<pat::ElectronCollection>      (iConfig.getParameter<edm::InputTag>("electrons"));         // slimmedElectrons
```
Comment (davidlange6) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc]: hi @archiron - please make the indentation here the same as the rest of the code.

File: Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc [hunk 2]
```diff
@@ -215,18 +216,16 @@ void ElectronMcSignalValidatorMiniAOD::analyze(const edm::Event& iEvent, const e
     iEvent.getByToken(mcTruthCollection_, genParticles) ;  
 
     //recomp
-    edm::Handle <edm::ValueMap <float> > ValueMaps_ChargedHadrons;
-    edm::Handle <edm::ValueMap <float> > ValueMaps_NeutralHadrons;
-    edm::Handle <edm::ValueMap <float> > ValueMaps_Photons;
+    edm::Handle <edm::ValueMap <float> > pfSumChargedHadronPtTmp;
```
Comment (davidlange6) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc]: and also here.

File: Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h [hunk 3]
```diff
@@ -113,7 +113,7 @@ class ElectronMcSignalValidatorMiniAOD : public ElectronDqmAnalyzerBase {
 
 	MonitorElement *h1_ele_chargedHadronRelativeIso_mAOD_recomp;
 	MonitorElement *h1_ele_neutralHadronRelativeIso_mAOD_recomp;
-    MonitorElement *h1_ele_photonRelativeIso_mAOD_recomp;    
+    MonitorElement *h1_ele_photonRelativeIso_mAOD_recomp;
```
Comment (davidlange6) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h]: and here.

File: Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc (hunk 1 above)
Comment (archiron) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc]: this was done.

File: Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h (hunk 3 above)
Comment (archiron) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.h]: seems to be done

File: Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc (hunk 2 above)
Comment (archiron) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc]: Sorry, I don't see the pbm with indentation for lines 218-220.

**Comment by archiron** [comment]
I don't know why the indentation is not correct. When I look my files ( ) all is correct.

File: Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc (hunk 1 above)
Comment (davidlange6) [inline Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc]: doesn't seem to be the case - look here (clicking the "view" button):

https://github.com/archiron/cmssw/blob/7926ced23e66b1dfeaef7a98b5e30c66cdd4d9fd/Validation/RecoEgamma/plugins/ElectronMcMiniAODSignalValidator.cc

**Comment by dmitrijus** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**